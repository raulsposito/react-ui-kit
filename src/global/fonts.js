import { css } from 'styled-components'

export const WEIGHT = {
  XLIGHT: 200,
  LIGHT: 300,
  REGULAR: 400,
  SEMIBOLD: 600,
  BOLD: 700,
  XBOLD: 800
}

export const SIZE = {
  XSMALL: '10px',
  SMALL: '12px',
  MEDIUM: '14px',
  LARGE: '16px',
  XLARGE: '20px',
  XXLARGE: '24px',
  H_MEDIUM: '32px',
  H_LARGE: '40px',
  H_XLARGE: '48px'
}

export const xlight = css`font-weight: ${WEIGHT.XLIGHT};`
export const light = css`font-weight: ${WEIGHT.LIGHT};`
export const regular = css`font-weight: ${WEIGHT.REGULAR};`
export const semibold = css`font-weight: ${WEIGHT.SEMIBOLD};`
export const bold = css`font-weight: ${WEIGHT.BOLD};`
export const xbold = css`font-weight: ${WEIGHT.XBOLD};`

export const fontWeight = {
  xlight,
  light,
  regular,
  semibold,
  bold,
  xbold
}

export const FONT_WEIGHT = {
  XLIGHT: xlight,
  LIGHT: light,
  REGULAR: regular,
  SEMIBOLD: semibold,
  BOLD: bold,
  XBOLD: xbold
}

export const xsmall = css`font-size: ${SIZE.XSMALL};`
export const small = css`font-size: ${SIZE.SMALL};`
export const medium = css`font-size: ${SIZE.MEDIUM};`
export const large = css`font-size: ${SIZE.LARGE};`
export const xlarge = css`font-size: ${SIZE.XLARGE};`
export const xxlarge = css`font-size: ${SIZE.XXLARGE};`

export const hmedium = css`font-size: ${SIZE.H_MEDIUM};`
export const hlarge = css`font-size: ${SIZE.H_LARGE};`
export const hxlarge = css`font-size: ${SIZE.H_XLARGE};`

export const fontSize = {
  xsmall,
  small,
  medium,
  large,
  xlarge,
  xxlarge,

  hmedium,
  hlarge,
  hxlarge
}

export const FONT_SIZE = {
  XSMALL: xsmall,
  SMALL: small,
  MEDIUM: medium,
  LARGE: large,
  XLARGE: xlarge,
  XXLARGE: xxlarge,
  HMEDIUM: hmedium,
  HLARGE: hlarge,
  HXLARGE: hxlarge
}

export const openSans = css`font-family: 'Open Sans', sans-serif;`
export const roboto = css`font-family: 'Roboto', sans-serif;`
