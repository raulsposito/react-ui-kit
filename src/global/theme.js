// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
