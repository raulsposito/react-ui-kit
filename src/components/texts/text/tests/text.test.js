import React from 'react'
import { render, screen } from '@testing-library/react'

import { SIZE, WEIGHT } from '@global/fonts'
import { colors, spacing } from '@global/theme'

import Text from '../index'

describe('<Text />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Text text='Goku' />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Children & text', () => {
    it('has children', () => {
      render(<Text>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toBeInTheDocument()
    })
    it('has prop text', () => {
      render(<Text text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toBeInTheDocument()
    })
  })

  describe('Disabled', () => {
    it('is not disabled', () => {
      render(<Text text='Goku' />)
      const t = screen.getByText('Goku')

      expect(t).toHaveStyle({ color: `${colors.contrast.regular}` })
    })

    it('is not disabled and have color', () => {
      render(<Text text='Goku' color='variant' />)
      const t = screen.getByText('Goku')

      expect(t).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('is disabled', () => {
      render(<Text text='Goku' disabled />)
      const t = screen.getByText('Goku')

      expect(t).toHaveAttribute('disabled')
      expect(t).toHaveStyle({ color: `${colors.contrast.disabled}` })
    })

    it('is disabled and have color', () => {
      render(<Text text='Goku' color='variant' disabled />)
      const t = screen.getByText('Goku')

      expect(t).toHaveAttribute('disabled')
      expect(t).toHaveStyle({ color: `${colors.variant.disabled}` })
    })
  })

  describe('Sizes', () => {
    it('xsmall', () => {
      render(<Text text='Goku' size='xsmall' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.XSMALL}` })
    })

    it('small', () => {
      render(<Text text='Goku' size='small' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.SMALL}` })
    })

    it('medium', () => {
      render(<Text text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.MEDIUM}` })
    })

    it('large', () => {
      render(<Text text='Goku' size='large' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.LARGE}` })
    })

    it('xlarge', () => {
      render(<Text text='Goku' size='xlarge' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.XLARGE}` })
    })

    it('xxlarge', () => {
      render(<Text text='Goku' size='xxlarge' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-size': `${SIZE.XXLARGE}` })
    })
  })

  describe('Weight', () => {
    it('xlight', () => {
      render(<Text weight='xlight'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.XLIGHT}` })
    })

    it('light', () => {
      render(<Text weight='light'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.LIGHT}` })
    })

    it('regular', () => {
      render(<Text>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.REGULAR}` })
    })

    it('semibold', () => {
      render(<Text weight='semibold'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.SEMIBOLD}` })
    })

    it('bold', () => {
      render(<Text weight='bold'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.BOLD}` })
    })

    it('xbold', () => {
      render(<Text weight='xbold'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'font-weight': `${WEIGHT.XBOLD}` })
    })
  })

  describe('Colors', () => {
    it('contrast', () => {
      render(<Text text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.contrast.regular}` })
    })

    it('primary', () => {
      render(<Text color='primary' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.primary.regular}` })
    })

    it('secondary', () => {
      render(<Text color='secondary' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.secondary.regular}` })
    })

    it('variant', () => {
      render(<Text color='variant' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('success', () => {
      render(<Text color='success' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.success.regular}` })
    })

    it('warning', () => {
      render(<Text color='warning' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.warning.regular}` })
    })

    it('error', () => {
      render(<Text color='error' text='Goku' />)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ color: `${colors.error.regular}` })
    })
  })

  describe('Align', () => {
    it('aligns left by default', () => {
      render(<Text>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'text-align': 'left' })
    })

    it('aligns left', () => {
      render(<Text align='left'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'text-align': 'left' })
    })

    it('aligns right', () => {
      render(<Text align='right'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'text-align': 'right' })
    })

    it('aligns center', () => {
      render(<Text align='center'>Goku</Text>)
      const t = screen.getByText('Goku')
      expect(t).toHaveStyle({ 'text-align': 'center' })
    })
  })

  describe('Margin Bottom', () => {
    it('has margin bottom 0 if no prop is passed', () => {
      render(<Text>Spiderman</Text>)
      const h = screen.getByText('Spiderman')
      expect(h).toHaveStyle({ 'margin-bottom': '0' })
    })

    it('has margin bottom 8px if one is passed', () => {
      render(<Text marginBottom='one'>Spiderman</Text>)
      const h = screen.getByText('Spiderman')
      expect(h).toHaveStyle({ 'margin-bottom': `${spacing.one}` })
    })

    it('adds a margin bottom to a value that is not valid', () => {
      render(<Text marginBottom='fifty'>Spiderman</Text>)
      const h = screen.getByText('Spiderman')
      expect(h).toHaveStyle({ 'margin-bottom': '0' })
    })
  })
})
