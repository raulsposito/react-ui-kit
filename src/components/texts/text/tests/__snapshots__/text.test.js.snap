// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match snapshot 1`] = `
<DocumentFragment>
  <p
    class="sc-bdfBQB dyFnKw"
  >
    Goku
  </p>
</DocumentFragment>
`;
